core/include/drivers/clk.h

```diff
@@ -168,4 +171,16 @@ struct clk *clk_get_parent_by_index(struct clk *clk, size_t pidx);
  */
 TEE_Result clk_set_parent(struct clk *clk, struct clk *parent);
 
+/**
+ * clk_get_rates_array - Get supported rates as an array
+ *
+ * @clk: Clock for which the rates are requested
+ * @start_index: start index of requested rates
+ * @rates: Array of rates allocated by caller or NULL to query count of rates
+ * @nb_elts: Max number of elements that the array can hold
```

Comment:
this descries the input value.
output value is the actual return rate array size.